Sourcery/main.swift

```diff
@@ -67,8 +67,22 @@ fileprivate enum Validators {
         return path
     }
 
-    static func isWriteable(path: Path) -> Path {
-        if path.exists && !path.isWritable {
+    static func isWritable(path: Path, autoLock: Bool) -> Path {
+        let pathExistsAndIsNotWritable = path.exists && !path.isWritable
+        let pathExistsAndWillNotBeWritableIfUnlocked: Bool = {
+            guard path.exists, autoLock, path.isImmutable else {
+                return false
+            }
+            
+            try! path.setImmutable(false)
```

Comment:
The idea here is checking if the file would be writable if it were unlocked so we unlock it, check if it's writable and lock it again to try to avoid any side effects